internal/validation/gateway/httproute_test.go

```diff
@@ -255,113 +257,6 @@ func TestValidateHTTPRoute(t *testing.T) {
 			validationMsg: "httproute spec did not pass validation",
 			err:           fmt.Errorf("queryparam matching is not yet supported for httproute"),
 		},
-		{
-			msg: "if an HTTPRoute is using regex path matching it fails validation due to lack of support",
```

Comment:
While removing this: do we have a test case here that would have a route with regex path  which passes the test? 🤔

Reply:
Oh is it now in `commonHTTPRouteValidationTestCases` ?

Reply:
That validation has been removed in this PR because it had been supported since closing
- https://github.com/Kong/kubernetes-ingress-controller/issues/2153
It's somewhat surprising that no user has noticed it.

Now it's allowed and covered with a test defined in `commonHTTPRouteValidationTestCases` 
https://github.com/Kong/kubernetes-ingress-controller/blob/39ab171288ca6bc994ed1f4fe4bddcf6746ce3c7/test/integration/httproute_webhook_test.go#L92-L117